iroh/src/protocol.rs

```diff
@@ -268,7 +268,10 @@ impl RouterBuilder {
         let mut join_set = JoinSet::new();
         let endpoint = self.endpoint.clone();
         let protos = protocols.clone();
-        let cancel = CancellationToken::new();
+
+        // We use a child token of the endpoint, to ensure that this is shutdown
+        // when the endpoint is shutdown, but that we can shutdown ourselves independently.
+        let cancel = endpoint.cancel_token().child_token();
```

Comment:
So you can shutdown a router without shutting down the endpoint. Makes sense I guess.

Nobody is stopping you from running multiple routers on the same endpoint, but that would probably be a pretty bad idea...

Reply:
we currently shutdown the endpoint as well from the router, but we could change that